src/mdns/mdns_avahi.cpp

```diff
@@ -590,13 +636,47 @@ otbrError PublisherAvahi::UnpublishService(const char *aName, const char *aType)
 
 otbrError PublisherAvahi::PublishHost(const char *aName, const uint8_t *aAddress, uint8_t aAddressLength)
 {
-    OTBR_UNUSED_VARIABLE(aName);
-    OTBR_UNUSED_VARIABLE(aAddress);
-    OTBR_UNUSED_VARIABLE(aAddressLength);
+    otbrError    ret   = OTBR_ERROR_NONE;
+    int          error = 0;
+    char         fullHostName[kMaxSizeOfHost];
+    Host *       host;
+    AvahiAddress address;
 
-    VerifyOrDie(false, "PublishHost is not implemented with avahi");
+    VerifyOrExit(mState == State::kReady, errno = EAGAIN, ret = OTBR_ERROR_ERRNO);
+    VerifyOrExit(aName != nullptr, ret = OTBR_ERROR_INVALID_ARGS);
+    VerifyOrExit(aAddress != nullptr, ret = OTBR_ERROR_INVALID_ARGS);
+    VerifyOrExit(aAddressLength == sizeof(address.data.ipv6.address), ret = OTBR_ERROR_INVALID_ARGS);
 
-    return OTBR_ERROR_NONE;
+    SuccessOrExit(ret = MakeFullName(fullHostName, sizeof(fullHostName), aName));
+    SuccessOrExit(ret = GetOrCreateHost(mClient, aName, &host));
+    // Check if the update is needed
+    VerifyOrExit(!host->mIsCommitted || memcmp(host->mAddress.data.ipv6.address, aAddress, aAddressLength));
+
+    address.proto = AVAHI_PROTO_INET6;
+    memcpy(&address.data.ipv6.address[0], aAddress, aAddressLength);
+
+    otbrLog(OTBR_LOG_INFO, "MDNS set address for host %s", fullHostName);
+    error = avahi_entry_group_add_address(host->mGroup, AVAHI_IF_UNSPEC, AVAHI_PROTO_INET6, AVAHI_PUBLISH_NO_REVERSE,
```

Comment:
We need to remove existing addresses. Current `Publisher::PublishHost` API has the semantic of `SET` but not `ADD`. Although multiple addresses can be associated with a single hostname, current API doesn't support it yet.

Reply:
Given that Avahi forbids modifying a committed entry group, in order to override an existing address we would need to store all service information in the Publisher class itself and rebuild the Avahi entry group from scratch every time any modification is needed. Do you think it would be a good approach?